hydra-node/test/Hydra/API/RestServerSpec.hs

```diff
@@ -17,3 +21,27 @@ spec = parallel $ do
 
   roundtripAndGoldenSpecs
     (Proxy @(ReasonablySized (DraftCommitTxRequest Tx)))
+
+  aroundAll withJsonSpecifications $
+    -- XXX: It's unintuitive that dir/api.yaml is in fact a JSON file holding
+    -- the JSON schema!
+    specify "Validate /commit publish api schema" $ \dir -> do
+      property $
+        withMaxSuccess 1 $ do
+          conjoin
+            [ prop_validateJSONSchema @(DraftCommitTxRequest Tx) (dir </> "api.yaml") (key "channels" . key "/commit" . key "publish" . key "message" . key "payload")
+            , prop_specIsComplete @(ReasonablySized (DraftCommitTxRequest Tx)) (dir </> "api.yaml") apiSpecificationSelector
+            ]
+  aroundAll withJsonSpecifications $
+    -- XXX: It's unintuitive that dir/api.yaml is in fact a JSON file holding
+    -- the JSON schema!
+    specify "Validate /commit subscribe api schema" $ \dir -> do
+      property $
+        withMaxSuccess 1 $ do
+          conjoin
+            [ prop_validateJSONSchema @(DraftCommitTxResponse Tx) (dir </> "api.yaml") (key "channels" . key "/commit" . key "subscribe" . key "message" . key "payload")
+            , prop_specIsComplete @(ReasonablySized (DraftCommitTxResponse Tx)) (dir </> "api.yaml") apiSpecificationSelector
```

Comment:
In principle checking completeness of possible messages on the response side would make sense, this property is nonsense given the current `api.yaml`.

See also my comment that we should enumerate the error responses in the api specification.